src/unitxt/metrics.py

```diff
@@ -400,31 +699,180 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
+                group_to_instance_scores[group_key][field_name][idx].append(
+                    instance["score"]["instance"][field_name]
+                )
+
+        # now for each group, take the aggregation function, comparing others to baseline
+        return [
+            {
+                "score": {
+                    "instance": {
+                        field_name: group_aggregation_func(
+                            baseline_scores=scores[field_name][0],
+                            other_scores=scores[field_name][1],
+                        )
+                        for field_name in field_names
+                    }
+                }
+            }
+            for scores in group_to_instance_scores.values()
+        ]
+
+    def _set_up_group_mean_aggregation(
+        self, instances, reduction_params, reduction_fields
+    ):
+        # if treat groups as units
+        if reduction_params["agg_func"][2]:
+            # pass the group aggregate---not instance---scores to resample as usual
+            aggregation_function = self.average_instance_scores
+            instances_to_resample = self.get_group_aggregated_instance_scores(
+                instances, reduction_fields, reduction_params["agg_func"][1]
+            )
+        else:
+            # pass the instance scores to resample, and calculate the group aggregation on the resamplings
+            instances_to_resample = deepcopy(instances)
+
+            def aggregation_function(
+                instances,
+                field_name,
+                group_aggregation_func=reduction_params["agg_func"][1],
+            ):
+                group_scores = self.get_group_aggregated_instance_scores(
+                    instances, [field_name], group_aggregation_func
+                )
+                return nan_mean(
+                    [group["score"]["instance"][field_name] for group in group_scores]
+                )
+
+        return instances_to_resample, aggregation_function
+
+    def _set_up_group_mean_subgroup_comparison_aggregation(
+        self, instances, reduction_params, reduction_fields
+    ):
+        # if treat groups as units
+        if reduction_params["agg_func"][2]:
+            # pass the group aggregate---not instance---scores to resample as usual
+            aggregation_function = self.average_instance_scores
+            instances_to_resample = (
+                self.get_group_aggregated_instance_scores_by_subgroups(
+                    instances,
+                    reduction_fields,
+                    group_aggregation_func=reduction_params["agg_func"][1],
+                    subgroup_field=reduction_params["subgroups"][0],
+                    baseline_name=reduction_params["subgroups"][1],
+                )
+            )
+        else:
+            # pass the instance scores to resample, and calculate the group aggregation on the resamplings
+            instances_to_resample = deepcopy(instances)
```

Comment:
```suggestion
            items_to_resample = deepcopy(instances)
```

Reply:
at this point its either groups or instances so i suggest to rename..

Reply:
or, maybe extract the instance scores here, and call them `scores_to_resample`. 
Since in the upper section of the if statements its group scores.